File: examples/rust/client_usage.rs
use dg_xch_core::blockchain::sized_bytes::Bytes32;
use dg_xch_core::traits::SizedBytes;
use julia_web_sdk::signature_client::SignatureClient;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = SignatureClient::with_url("https://example.com");

    let request_id = client.get_signature_request_id().await?;
    println!("request id: {request_id}");

    let status = client.get_signature_status().await?;
    println!("signature status: {status}");

    let nonce = Bytes32::new([0u8; 32]);
    let presentation = client
        .generate_signature_presentation(&request_id, nonce)
        .await?;
    println!(
        "presentation bytes: {}",
        presentation.compressed_presentation.len()
    );

    // Placeholder for local testing.
    // In production, pass the signed bytes returned by not.bot.
    client.verify_signature_presentation(&request_id, vec![]).await?;
    println!("verification request submitted");

    Ok(())
}

File: examples/javascript/client_usage.mjs
import { SignatureClient } from "../../javascript/src/index.js";
// App code should import from the package:
// import { SignatureClient } from "julia_web_sdk";

const client = new SignatureClient({
  baseUrl: process.env.JULIA_WEB_BASE_URL ?? "https://example.com"
});

async function main() {
  const requestId = await client.getSignatureRequestId();
  console.log("request id", requestId);

  const status = await client.getSignatureStatus();
  console.log("signature status", status);

  // The nonce and presentation values are placeholders.
  // Replace with real values produced by your not.bot flow.
  const nonce = "0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff";
  const presentationResponse = await client.generateSignaturePresentation(requestId, nonce);
  console.log("presentation response", presentationResponse);

  await client.verifySignaturePresentation(requestId, []);
  console.log("verification request submitted");
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
